### PR TITLE
Counter offer

### PR DESCRIPTION
**Pull request by Preport**

### CODE REVIEW

File: src/classes/Trades.ts [hunk 1]
```diff
@@ -675,6 +655,291 @@ export default class Trades {
         });
     }
 
+    private counterOffer(offer: TradeOffer): Promise<string> {
+        return new Promise((resolve, reject) => {
+            const counter = offer.counter();
+            // To the person who thinks about changing it. I have a gun keep out ( う-´)づ︻╦̵̵̿╤── \(˚☐˚”)/
+            counter.setMessage(
+                "Oni-chan. I'm not a dummie (thicc) offer contains wrong value.\nYou've probably made a few mistakes here's the correct offer."
+            );
```
Comment (idinium96) [inline src/classes/Trades.ts:664]: KEKW 🤣

Comment (Preport) [reply src/classes/Trades.ts:664]: Bruw KEKW

**Comment by idinium96** [comment]
Isn't better to put all the functions in the `counterOffer` method outside the `Trades` class, or as the other methods of `Trades` class?

**Comment by Preport** [comment]
I don't know probably. I made it like this cause you were making it inside the applyActionToOffer. So I thought you'd want it somewhere here ¯\_(ツ)_/¯.

**Comment by idinium96** [comment]
> I don't know probably. I made it like this cause you were making it inside the applyActionToOffer. So I thought you'd want it somewhere here ¯_(ツ)_/¯.

Hahaha. It's better to exercise the best practices. 😄

**Comment by Preport** [comment]
Yes I always do that. **\*Sweats nervously with 5000 lines single file projects\***

**Comment by idinium96** [comment]
Merging this into the `counter-offer` branch. Thanks a bunch!

**Review by idinium96 — APPROVED** [review]

**Pull request merged**